modules/service/src/main/scala/lucuma/odb/graphql/AttachmentRoutes.scala

```diff
@@ -42,105 +43,112 @@ object ObsAttachmentRoutes {
     maxUploadMb:           Int,
   ): HttpRoutes[F] =
     apply(
-      [A] => (u: User) => (fa: ObsAttachmentFileService[F] => F[A]) => pool.map(Services.forUser(u, enums, None)).map(_.obsAttachmentFileService(s3)).use(fa),
+      [A] => (u: User) => (fa: AttachmentFileService[F] => F[A]) => pool.map(Services.forUser(u, enums, None)).map(_.attachmentFileService(s3)).use(fa),
       ssoClient,
       maxUploadMb
     )
 
   // used by tests
   def apply[F[_]: Async: Trace](
-    service:     ObsAttachmentFileService[F],
+    service:     AttachmentFileService[F],
     ssoClient:   SsoClient[F, User],
     maxUploadMb: Int,
   ): HttpRoutes[F] =
     apply(
-      [A] => (u: User) => (fa: ObsAttachmentFileService[F] => F[A]) => fa(service),
+      [A] => (u: User) => (fa: AttachmentFileService[F] => F[A]) => fa(service),
       ssoClient,
       maxUploadMb
     )
 
 
   def apply[F[_]: Concurrent: Trace: UUIDGen](
-    service:               [A] => User => (ObsAttachmentFileService[F] => F[A]) => F[A],
+    service:               [A] => User => (AttachmentFileService[F] => F[A]) => F[A],
     ssoClient:             SsoClient[F, User],
     maxUploadMb:           Int
   ): HttpRoutes[F] = {
     val dsl = Http4sDsl[F]
     import dsl._
 
     extension (exc: AttachmentException)
-      def toResponse: F[Response[F]] = exc match {
+      def toErrorResponse: F[Response[F]] = exc match {
         case AttachmentException.Forbidden        => Forbidden()
         case AttachmentException.FileNotFound     => NotFound()
         case AttachmentException.InvalidRequest(msg) => BadRequest(msg)
       }
 
-    given QueryParamDecoder[Tag] = QueryParamDecoder[String].map(s => Tag(s.toLowerCase))
+    extension[A](fe: F[Either[AttachmentException, A]])
+      def toResponse(fa: A => F[Response[F]]): F[Response[F]] =
+        fe.flatMap { _ match {
+          case Right(a) => fa(a)
+          case Left(e) => e.toErrorResponse
+        }
+        }
+
```

Comment:
FYI you actually don't need the `match` here; you can use partial function syntax where a total function is expected; i.e., 

```scala
      def toResponse(fa: A => F[Response[F]]): F[Response[F]] =
        fe.flatMap {
          case Right(a) => fa(a)
          case Left(e) => e.toErrorResponse
        }
```

Reply:
Thanks, I think my brain just got over full. 😆 I'll fix that and the formatting in another PR I'll have soon.